generative/2d_ldm/train_autoencoder.py

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+import sys
+from pathlib import Path
+
+import torch
+from generative.losses import PatchAdversarialLoss, PerceptualLoss
+from generative.networks.nets import PatchDiscriminator
+from monai.config import print_config
+from monai.utils import set_determinism
+from torch.nn import L1Loss, MSELoss
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.tensorboard import SummaryWriter
+
+from utils import KL_loss, define_instance, prepare_brats2d_dataloader, setup_ddp
+from visualize_image import visualize_2d_image
+
+
+def main():
+    parser = argparse.ArgumentParser(description="PyTorch Object Detection Training")
+    parser.add_argument(
+        "-e",
+        "--environment-file",
+        default="./config/environment.json",
+        help="environment json file that stores environment path",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        default="./config/config_train_48g.json",
+        help="config json file that stores hyper-parameters",
+    )
+    parser.add_argument(
+        "--download_data",
+        default=False,
+        action="store_true",
+        help="whether to download Brats data before training",
+    )
+    parser.add_argument("-g", "--gpus", default=1, type=int, help="number of gpus per node")
+    args = parser.parse_args()
+
+    # Step 0: configuration
+    ddp_bool = args.gpus > 1  # whether to use distributed data parallel
+    if ddp_bool:
+        rank = int(os.environ["LOCAL_RANK"])
+        world_size = int(os.environ["WORLD_SIZE"])
+        dist, device = setup_ddp(rank, world_size)
+    else:
+        rank = 0
+        world_size = 1
+        device = 0
+
+    torch.cuda.set_device(device)
+    print(f"Using {device}")
+
+    print_config()
+    torch.backends.cudnn.benchmark = True
+    torch.set_num_threads(4)
+    torch.autograd.set_detect_anomaly(True)
+
+    env_dict = json.load(open(args.environment_file, "r"))
+    config_dict = json.load(open(args.config_file, "r"))
+
+    for k, v in env_dict.items():
+        setattr(args, k, v)
+    for k, v in config_dict.items():
+        setattr(args, k, v)
+
+    set_determinism(42)
+    os.environ.get("MONAI_DATA_DIRECTORY")
```

Comment:
@Can-Zhao This seems redundant.

Like the 3D LDM, since it is not possible to change to `MONAI_DATA_DIRECTORY`, we should remove this from the script

Reply:
Thank you! Sorry I did not update it